src/AlertBar/index.js

```diff
@@ -0,0 +1,132 @@
+import React, { PureComponent } from 'react'
+import propTypes from 'prop-types'
+import cx from 'classnames'
+import styles, { ANIMATION_TIME } from './styles'
+
+import { Actions, actionsPropType } from './Actions'
+import { Dismiss } from './Dismiss'
+import { Icon, iconPropType } from './Icon'
+import { Message } from './Message'
+import { mutuallyExclusive } from '../prop-validators'
+
+class AlertBar extends PureComponent {
+    state = {
+        visible: false,
+        hidden: false,
+    }
+
+    componentDidMount() {
```

Comment:
I think this stuff should go into it's own function that's called both when the component did mount as well as when the component did update && the visible prop changed to true (in case someone using this library is rendering error messages (like for a form) and shows/hides/shows it (like a "you have to fill out all inputs" message for example)